console/frontend/pom.xml

```diff
@@ -9,22 +9,72 @@
 
 	<artifactId>ibis-adapterframework-console-frontend</artifactId>
 	<name>Frank Console Frontend</name>
-	<packaging>war</packaging>
-
-	<dependencies>
-		<dependency>
-			<groupId>org.ibissource</groupId>
-			<artifactId>ibis-adapterframework-core</artifactId>
-		</dependency>
-		<dependency>
-			<groupId>javax.servlet</groupId>
-			<artifactId>javax.servlet-api</artifactId>
-			<scope>provided</scope>
-		</dependency>
-	</dependencies>
 
 	<build>
+		<defaultGoal>process-resources</defaultGoal>
 		<plugins>
+			<plugin>
+				<groupId>com.github.eirslett</groupId>
+				<artifactId>frontend-maven-plugin</artifactId>
+				<version>1.12.1</version>
+				<executions>
+					<execution>
+						<id>download npm</id>
+						<phase>initialize</phase>
+						<goals>
+							<goal>install-node-and-npm</goal>
+						</goals>
+					</execution>
+					<execution>
+						<id>run npm install</id>
+						<phase>initialize</phase>
+						<goals>
+							<goal>npm</goal>
+						</goals>
+						<configuration>
+							<arguments>install</arguments>
+						</configuration>
+					</execution>
+
+					<execution>
+						<id>run npm build</id>
+						<phase>generate-resources</phase>
+						<goals>
+							<goal>npm</goal>
+						</goals>
+						<configuration>
+							<arguments>run build:prod</arguments>
```

Comment:
Eigenlijk zouden we moeten kijken of we in een development of productie omgeving zitten en afhankelijk daarvan `build` of `build:prod` te runnen, nu weet ik niet of dat goed mogelijk is met de plugin maar het zou de build tijd wel moeten verminderen / halveren tijdens development

Reply:
Ik denk dat je tijdens het ontwikkelen eigenlijk `ng serve` zou willen doen, en de backend aan willen laten staan, zodat je direct je aanpassingen kan zien.

Reply:
Voor de nieuwe angular is dit inderdaad een goede optie, voor angularjs + webpack zou je wel webpack serve kunnen doen maar ik weet niet tot hoever er een ingebouwde proxy is voor de api calls zoals je die in de nieuwe angular wel kan hebben.

Overigens is er ook nog de issue van base urls waar we naar moeten gaan kijken, maar dat is vooral belangrijk voor de nieuwe angular (Ik kan me herinneren dat ik via de index jsp een goede kloppende baseurl moest opgeven maar nu gebruiken we geen jsp meer terwijl de application context wel dynamisch blijft)